docs/docs/tutorial/part-4/index.mdx

```diff
@@ -0,0 +1,918 @@
+      </p>
+      <p style={{ marginBottom: 0 }}>
+        If you{"'"}re looking for a full experience in the meantime, you can check out the (slightly outdated){" "}
+        <a href="https://v2.gatsbyjs.com/docs/tutorial/">
+          <strong>Gatsby v2 Tutorial</strong>
+        </a>
+        .
+      </p>
+    </div>
+  }
+/>
+
+## Introduction
+
+Parts 4 and 5 of this Tutorial are all about data!
+
+So far, you've been writing text and adding images directly in your React components. That's an excellent way to build many websites! But often it's easier to create and maintain data somewhere else - like a folder of Markdown files or a content management system (CMS) - and then pull it into your components as needed. That way, you can make updates to your content without affecting the code for your site.
+
+Conveniently, Gatsby has a powerful feature called the **data layer** that you can use to pull data into your site from anywhere. Want to keep your blog posts in WordPress, your store products in Shopify, and your user data in AirTable? No problem! With Gatsby's data layer, you can combine data from multiple sources, which lets you choose the best platform for each type of data.
+
+<Announcement style={{marginBottom: "1.5rem"}}>
+
+Gatsby's data layer is powered by a technology called **GraphQL**. GraphQL is a query language with a special syntax that lets you ask for the data you need inside a component.
+
+In this Tutorial, we'll teach you all the GraphQL you'll need to know to build your first Gatsby site. Interested in learning more? [How To GraphQL](https://www.howtographql.com/) is a free tutorial that teaches you the fundamentals.
+
+</Announcement>
+
+In this part of the Tutorial, you'll learn about how to add data to Gatsby's data layer and how to pull that data into your React components.
+
+By the end of this part of the Tutorial, you will be able to:
+
+- Use **GraphiQL** to explore the data in the data layer and build your own GraphQL queries.
+- Use the **`useStaticQuery`** hook to pull data into a "building-block" component.
+- Use the **`gatsby-source-filesystem`** plugin to pull data into your site from your computer's filesystem.
+- Create a **page query** to pull data into a page component.
+
+## Meet Gatsby's GraphQL data layer
+
+Gatsby has its own [GraphQL](https://graphql.org/) data layer where it keeps all the data for your site. But how does it work under the hood?
+
+First, your data is stored in one or more **sources**. That source might be a folder on your computer's filesystem, a content management system (CMS) like WordPress, or a database. You might even have multiple data sources!
+
+How do you get from its source into the data layer? By adding special plugins to your site called **source plugins**. Each source plugin is designed to communicate with a specific source. When you build your site, each source plugin pulls data from its particular source and adds it to the GraphQL data layer for your site.
```

Comment:
I personally wouldn't say that source plugins are "special". I'd argue that themes are "special" because they are technically plugins but are used a little bit different than source plugins, transformer plugins

Reply:
Hmm good point. Thanks!

```suggestion
How do you get from its source into the data layer? By adding a type of plugin to your site called a **source plugin**. Each source plugin is designed to communicate with a specific source. When you build your site, each source plugin pulls data from its particular source and adds it to the GraphQL data layer for your site.
```